src/lib/Layout/PanelSideBarLayout/PanelSideBar/Context/PanelSideBarContext.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { ComponentType, createContext, ReactNode, useContext, useState } from "react";
+import { PanelItem } from "../Definitions/PanelItem";
+
+export interface PanelLinkRendererProps<T> {
+  item: PanelItem<T>;
+  children: ReactNode;
+}
+
+export type MenuItemToggleFn<TPanelItem> = (menuItem: PanelItem<TPanelItem>) => void;
+
+export interface PanelSideBarContextProps<TPanelItem> {
+  activePanelId: string;
+  /**
+   * The global panel items.
+   */
+  globalItems: PanelItem<TPanelItem>[];
+  /**
+   * The local panel items.
+   */
+  localItems?: PanelItem[];
+  /**
+   * The component used to render the menu item links.
+   */
+  setActivePanel: (panelId: string) => void;
+  LinkRenderer: ComponentType<PanelLinkRendererProps<TPanelItem>>;
```

Comment:
i think this props are exposed too right? as they are not commented properly yet..

Reply:
@neotob Exposed props commented

Reply:
awesome thank you!